pkg/iac/terraform/state/enumerator/s3.go

```diff
@@ -0,0 +1,58 @@
+package enumerator
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/cloudskiff/driftctl/pkg/iac/config"
+	"github.com/pkg/errors"
+)
+
+type S3EnumeratorConfig struct {
+	Bucket *string
+	Prefix *string
+}
+
+type S3Enumerator struct {
+	config config.SupplierConfig
+	client s3iface.S3API
+}
+
+func NewS3Enumerator(config config.SupplierConfig) (*S3Enumerator, error) {
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+	return &S3Enumerator{
+		config,
+		s3.New(sess),
+	}, nil
+}
+
+func (s *S3Enumerator) Enumerate() ([]string, error) {
+	bucketPath := strings.Split(s.config.Path, "/")
+	if len(bucketPath) < 2 {
+		return nil, errors.Errorf("Unable to parse S3 path: %s. Must be BUCKET_NAME/PREFIX", s.config.Path)
+	}
+	bucket := bucketPath[0]
+	prefix := strings.Join(bucketPath[1:], "/")
+
+	keys := make([]string, 0)
+	input := &s3.ListObjectsV2Input{
+		Bucket: &bucket,
+		Prefix: &prefix,
+	}
+	err := s.client.ListObjectsV2Pages(input, func(output *s3.ListObjectsV2Output, lastPage bool) bool {
```

Comment:
Hmm it's adding dependency to the remote pkg, I don't think we should use repositories outside remote pkg for now. IaC and remote pkg should not include themselves for me.

If we want to do this we need to move `AwsClientFactory` and other repositories outside remote pkg.

It could be a nice thing to always use repositories to access cloud providers, but it will require some refactoring imho.

BTW the actual s3 backend is not using any repo (I used the same approach for enumerators)

WDYT ?

Reply:
I think it would be nice to use repository everywhere but I get that the refacto does not have its place in this PR no problem